addons/web/static/tests/core/datetime/datetime_input_tests.js

```diff
@@ -503,4 +504,42 @@ QUnit.module("Components", ({ beforeEach }) => {
 
         assert.strictEqual(input.value, "١٥ يوليو, ٢٠٢٠ ١٢:٣٠:٤٣");
     });
+
+    QUnit.test("check parseDate in localization with textual month format", async function (assert) {
+        Object.assign(localization, {
+            dateFormat: 'MMM/dd/yyyy',
+            timeFormat: 'HH:mm:ss',
+            dateTimeFormat: 'MMM/dd/yyyy HH:mm:ss',
+        });
+
+        // The following assertions should all yield the same result
+        assert.strictEqual(parseDate('Jan/05/1997').toFormat('dd/MM/yyyy'), '05/01/1997');
+        assert.strictEqual(parseDate('Jan/05/1997', { format: undefined }).toFormat('dd/MM/yyyy'), '05/01/1997');
```

Comment:
The test should be in `tests/core/l10n/dates_tests`. Only this assert is not tested, you can add a small test `parseDate with undefined format` near `parseDate with small notations`